geode-lucene/src/main/java/org/apache/geode/cache/lucene/internal/AbstractPartitionedRepositoryManager.java

```diff
@@ -111,6 +121,10 @@ protected IndexRepository computeRepository(Integer bucketId) {
     return repo;
   }
 
+  protected void allowRepositoryComputation() {
```

Comment:
I was under the impression that the latch is waiting before indexRepositories.compute in the computeRepositories method call hence the repo was not created yet. But the coutdown allows it to move ahead an compute the repository.